src/markdown/parser.cr

```diff
@@ -469,6 +481,40 @@ class Markdown::Parser
     str[pos].chr.whitespace?
   end
 
+  def starts_with_plus?(line)
+    bytesize = line.bytesize
+    str = line.to_unsafe
+    pos = 0
+    while pos < bytesize && str[pos].chr.whitespace?
+      pos += 1
+    end
+
+    return false unless pos < bytesize
+    return false unless str[pos].chr == '+'
+
+    pos += 1
+
+    return false unless pos < bytesize
+    str[pos].chr.whitespace?
+  end
+
+  def starts_with_hyphen?(line)
+    bytesize = line.bytesize
+    str = line.to_unsafe
+    pos = 0
+    while pos < bytesize && str[pos].chr.whitespace?
+      pos += 1
+    end
+
+    return false unless pos < bytesize
+    return false unless str[pos].chr == '-'
+
+    pos += 1
+
+    return false unless pos < bytesize
+    str[pos].chr.whitespace?
+  end
```

Comment:
Instead of duplicating the method three times, parameterize the token, `starts_with_bullet_list_marker?`, `starts_with_bullet_list_marker?('*')`